src/MongoDB.Analyzer/Core/Utilities/SymbolExtensions.cs

```diff
@@ -155,6 +156,9 @@ symbol switch
         _ => symbol.IsContainedInLambda(parentNode)
     };
 
+    public static bool IsDBSet(this ITypeSymbol typeSymbol) =>
+        typeSymbol?.ContainingNamespace?.ToDisplayString() == NamespaceEF;
```

Comment:
Sorry, I meant let's check for that it's a symbol of DBSet type, not just a type defined in NamespaceEF.